proto/tendermint/types/types.proto

```diff
@@ -87,6 +87,63 @@ message Data {
   // NOTE: not all txs here are valid.  We're just agreeing on the order first.
   // This means that block.AppHash does not include these txs.
   repeated bytes txs = 1;
+  IntermediateStateRoots intermediate_state_roots = 2 [(gogoproto.nullable) = false];
+  EvidenceData evidence    = 3 [(gogoproto.nullable) = false];
+  Messages messages        = 4 [(gogoproto.nullable) = false];
+}
+
+// DuplicateVoteEvidence contains evidence a validator signed two conflicting
+// votes.
+message DuplicateVoteEvidence {
+  Vote vote_a = 1;
+  Vote vote_b = 2;
+}
+
+message LightClientAttackEvidence {
+  LightBlock conflicting_block = 1;
+  int64      common_height     = 2;
+}
+
+message Evidence {
+  oneof sum {
+    DuplicateVoteEvidence     duplicate_vote_evidence      = 1;
+    LightClientAttackEvidence light_client_attack_evidence = 2;
+  }
+}
+
+// EvidenceData contains any evidence of malicious wrong-doing by validators
+message EvidenceData {
+  repeated Evidence evidence = 1 [(gogoproto.nullable) = false];
+}
+
+message IntermediateStateRoots {
+  repeated bytes raw_roots_list = 1;
+}
+
+message Messages {
+  repeated Message messages_list = 1;
+}
+
+message Message {
+  bytes namespace_id = 1;
+  bytes data = 2;
+}
+
+// DataAvailabilityHeader contains the row and column roots of the erasure
+// coded version of the data in Block.Data.
+// Therefor the original Block.Data is arranged in a
+// k × k matrix, which is then "extended" to a
+// 2k × 2k matrix applying multiple times Reed-Solomon encoding.
+// For details see Section 5.2: https://arxiv.org/abs/1809.09044
+// or the LazyLedger specification:
+// https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#availabledataheader
+// Note that currently we list row and column roots in separate fields
+// (different from the spec).
+message DataAvailabilityHeader {
+  // RowRoot_j 	= root((M_{j,1} || M_{j,2} || ... || M_{j,2k} ))
+  repeated bytes row_roots = 1;
+  // ColumnRoot_j = root((M_{1,j} || M_{2,j} || ... || M_{2k,j} ))
+  repeated bytes column_roots = 2;
 }
```

Comment:
@adlerjohn  can this be useful for the protobufs you wanted to add to the spac, too?

Reply:
Yeah, I can transclude most of the type definitions here I think.